src/main/java/com/zaxxer/nuprocess/osx/ProcessKqueue.java

```diff
@@ -192,59 +261,38 @@ else if (WIFSIGNALED(status)) {
             osxProcess.onExit(status);
          }
       }
-      else if (filter == Kevent.EVFILT_SIGNAL) {
-         checkStdinCloses();
-         checkWaitWrites();
-      }
-
-      kevent.clear();
-
-      return true;
    }
 
    // ************************************************************************
    //                             Private methods
    // ************************************************************************
 
-   private void queueEvent(int handle, int filter, int flags, int fflags, Pointer udata)
-   {
-      try {
-         Kevent kevent = keventPool.take();
-
-         Kevent.EV_SET(kevent, (long) handle, filter, flags, fflags, 0l, udata);
-         LibKevent.kevent(kqueue, kevent.getPointer(), 1, null, 0, null);
-
-         keventPool.put(kevent);
-      }
-      catch (InterruptedException e) {
-         return;
-      }
-   }
-
    private void checkStdinCloses()
    {
-      if (closeQueue.isEmpty()) {
-         return;
-      }
-
-      OsxProcess process;
-      while ((process = closeQueue.poll()) != null) {
-         process.stdinClose();
-      }
+     List<OsxProcess> processes = new ArrayList<OsxProcess>();
+     closeQueue.drainTo(processes);
```

Comment:
@bhamiltoncx You're right, ``ArrayBlockingQueue`` implements an atomic ``drainTo()``.  I'm fine leaving this, but put a comment to the effect of:
```java
// drainTo() is known atomic for ArrayBlockingQueue
```